pinot-core/src/main/java/com/linkedin/pinot/core/startree/v2/AggregationFunction.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.startree.v2;
+
+import java.io.IOException;
+import javax.annotation.Nonnull;
+import com.linkedin.pinot.common.data.FieldSpec;
+import com.linkedin.pinot.core.query.aggregation.function.AggregationFunctionType;
+
+
+/**
+ * Interface to be implemented by each aggregation function. Here 'R' is the datatype for raw data to aggregate.
+ *  and 'A' is the datatype it supports.
+ */
+public interface AggregationFunction<R, A> {
```

Comment:
We allow different type of **R**, so don't make R generic (e.g. HLL we allow number as well as bytes)